File: 1653. Minimum Deletions to Make String Balanced.go
func minimumDeletions(s string) int {
    n := len(s)
    a, b := 0, 0
    if s[0] == 'a' {
        b = 1
    } else {
        a = 1
    }
    for i := 1; i < n; i++ {
        if s[i] == 'a' {
			b = min(a, b+1)
        } else {
			b = min(a, b)
			a++
        }
    }
    return min(a, b)
}

File: 1653. Minimum Deletions to Make String Balanced.java
class Solution {
    public int minimumDeletions(String s) {
        int a = s.charAt(0) == 'a' ? 0 : 1;
        int b = 1-a;
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == 'a') {
                b = Math.min(a, b+1);
            } else {
                b = Math.min(a, b);
                a++;
            }
        }
        return Math.min(a, b);
    }
}